src/dump.c

```diff
@@ -1257,31 +1258,48 @@ static void jl_serialize_lambdas_from_mod(jl_serializer_state *s, jl_module_t *m
     }
 }
 
+// flatten the backedge map reachable from caller into callees
+static void jl_collect_backedges_to(jl_serializer_state *s, jl_method_instance_t *caller, jl_array_t *callees, arraylist_t *to_restore)
+{
+    jl_array_t **pcallee = (jl_array_t**)ptrhash_bp(&edges_map, (void*)caller),
+                *callee = *pcallee;
+    if (callee != HT_NOTFOUND) {
+        arraylist_push(to_restore, (void*)pcallee);
+        arraylist_push(to_restore, (void*)callee);
+        *pcallee = (jl_array_t*) HT_NOTFOUND;
+        jl_array_ptr_1d_append(callees, callee);
```

Comment:
`callee` sounds singular to me, so it's a bit confusing that it's an array. Could use a comment explaining what it's an array of, and/or a variable rename. Is it an array of things called by `caller`?